File: MC_Pi.py
from random import uniform

def piApproximation(num):
    inside = 0

    for _ in range(num):
        x = uniform(-1, 1)
        y = uniform(-1, 1)

        if x**2+y**2 < 1:
            inside += 1

    return 4*inside/num

if __name__ == "__main__":
    print(piApproximation(10**6))

File: MC_Pi.cpp
#include <iostream>
#include <random>
#include <time.h>

double piApproximation(int N) {
    std::default_random_engine generator{static_cast<long unsigned int>(time(0))};
    std::uniform_real_distribution<double> distribution(-1, 1);
    double x, y, inside;

    for (int i=0; i<N; i++) {
        x = distribution(generator);
        y = distribution(generator);
        if (x*x + y*y < 1) {
            inside++;
        }
    }

    return 4*inside/N;
}

int main() {
    double result;
    int N = 100000000;
    result = piApproximation(N);
    std:: cout << result;
    return 0;
}